src/overlays/actors/ovl_En_Test7/z_en_test7.c

```diff
@@ -47,209 +48,193 @@ ActorInit En_Test7_InitVars = {
     /**/ EnTest7_Draw,
 };
 
-void EnTest7_SetupAction(EnTest7* this, EnTest7ActionFunc actionFunc) {
-    this->actionFunc = actionFunc;
+void EnTest7_SetupPlayerCamFunc(EnTest7* this, EnTest7PlayerCamFunc playerCamFunc) {
+    this->playerCamFunc = playerCamFunc;
 }
 
-void func_80AF082C(EnTest7* this, EnTest7UnkFunc func) {
-    this->unk_1E58 = func;
+void EnTest7_SetupAction(EnTest7* this, EnTest7ActionFunc actionFunc) {
+    this->actionFunc = actionFunc;
 }
 
-void func_80AF0838(EnTest7Struct2* arg0) {
-    EnTest7Struct2* ptr = arg0;
+void EnTest7_InitFeathers(OwlWarpFeather* feathers) {
+    OwlWarpFeather* feather = feathers;
     s32 i;
 
-    for (i = 0, ptr = arg0; i < 100; i++, ptr++) {
+    for (i = 0, feather = feathers; i < OWL_WARP_NUM_FEATHERS; i++, feather++) {
         s32 pad;
 
-        ptr->unk_00 = 0;
-        ptr->unk_04 = 0;
-        ptr->unk_08.x = 0.0f;
-        ptr->unk_08.y = 0.0f;
-        ptr->unk_08.z = 0.0f;
-        ptr->unk_14 = 0.0f;
-        ptr->unk_18 = 0.0f;
-        ptr->unk_1C = 0.0f;
-        ptr->unk_20 = 0.0f;
-        ptr->unk_24 = 0.0f;
-        ptr->unk_28 = 0.0f;
-        ptr->unk_2C = 0.00001f;
-        ptr->unk_30.x = 0;
-        ptr->unk_30.y = 0;
-        ptr->unk_30.z = 0;
-        ptr->unk_36 = 0;
-        ptr->unk_38 = 0;
-        ptr->unk_3A = 0;
-    }
-}
-
-void func_80AF0984(EnTest7Struct2* arg0, Vec3f* arg1, s32 arg2) {
-    s16 sp26 = Rand_ZeroOne() * 0xFFFF;
-    f32 sp20;
-    f32 temp_f0;
-    f32 temp_f0_2;
-    f32 temp_f0_3;
-    f32 temp_f4;
+        feather->type = OWL_WARP_FEATHER_TYPE_DISABLED;
+        feather->unk_04 = 0;
+        feather->pos.x = 0.0f;
+        feather->pos.y = 0.0f;
+        feather->pos.z = 0.0f;
+        feather->velocity.x = 0.0f;
+        feather->velocity.y = 0.0f;
+        feather->velocity.z = 0.0f;
+        feather->accel.x = 0.0f;
+        feather->accel.y = 0.0f;
+        feather->accel.z = 0.0f;
+        feather->scale = 0.00001f;
+        feather->rot.x = 0;
+        feather->rot.y = 0;
+        feather->rot.z = 0;
+        feather->angularVelocity.x = 0;
+        feather->angularVelocity.y = 0;
+        feather->angularVelocity.z = 0;
+    }
+}
 
-    arg0->unk_08 = *arg1;
+void EnTest7_AddFeather(OwlWarpFeather* feather, Vec3f* pos, s32 isPosYRand) {
+    s16 yaw = Rand_ZeroOne() * 0xFFFF;
+    f32 velocityXZ;
 
-    if (arg2 != 0) {
-        Math_CosS(sp26);
-        arg0->unk_08.x = arg0->unk_08.x;
-        arg0->unk_08.y += (Rand_ZeroOne() * 100.0f) - 20.0f;
-        Math_SinS(sp26);
-        arg0->unk_08.z = arg0->unk_08.z;
+    feather->pos = *pos;
+
+    if (isPosYRand) {
+        feather->pos.x += Math_CosS(yaw) * 0.0f;
+        feather->pos.y += (Rand_ZeroOne() * 100.0f) - 20.0f;
+        feather->pos.z += Math_SinS(yaw) * 0.0f;
```

Comment:
oh wtf. How did you even came up with this?

Reply:
If you look right below this code, there is
```c
    feather->velocity.x = Math_CosS(yaw) * velocityXZ;
    feather->velocity.y = Rand_ZeroOne();
    feather->velocity.z = Math_SinS(yaw) * velocityXZ;
```
And both `Math_CosS` and `Math_SinS` do nothing except get a return value, so just calling them as it was before meant nothing. So I assumed it was like the code below except `velocityXZ = 0.0f`